src/Core/Manifest/NodeVisitor.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace SilverStripe\Core\Manifest;
+
+use PhpParser\Node;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitorAbstract;
+
+class NodeVisitor extends NodeVisitorAbstract
```

Comment:
How about `ManifestVistor`? I was thinking the NS provides enough context here (`SilverStripe\Core\Manifest\NodeVisitor`). I could rename it during import `use SilverStripe\Core\Manifest\NodeVisitor as ManifestNodeVisitor` for more clarity outside the class?

Any other suggestions welcome